src/app/components/FundWalletModal/useFundWallet.ts

```diff
@@ -45,6 +46,14 @@ export const useFundWallet = (open: boolean) => {
     }
   }, [isError, isSuccess, depositData]);
 
+  useEffect(() => {
+    if (isSuccess && depositData.result?.redirectUrl3ds && depositData.result?.paymentId) {
```

Comment:
It's a better practice to have this logic declared outside the useEffect and inside it only execute it.